Node.js package installation fails on Node 13.0.1

### Problem description

I believe this is the same issue in `pulumi/pulumi`: https://github.com/pulumi/pulumi/issues/3434

```
FROM node:13
RUN curl -fsSL https://get.pulumi.com | sh
RUN npm i @pulumi/kubernetes
```
```
Step 1/3 : FROM node:13
---> a76ab9b9033f
Step 2/3 : RUN curl -fsSL https://get.pulumi.com | sh
---> Using cache
---> 14d7e19487ed
Step 3/3 : RUN npm i @pulumi/kubernetes
---> Running in 19f50086fc95
> deasync@0.1.16 install /node_modules/deasync
> node ./build.js
`linux-x64-node-13` exists; testing
Binary is fine; exiting
> grpc@1.21.1 install /node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library
node-pre-gyp WARN Using needle for node-pre-gyp https download
node-pre-gyp WARN Tried to download(404): https://node-precompiled-binaries.grpc.io/grpc/v1.21.1/node-v79-linux-x64-glibc.tar.gz
node-pre-gyp WARN Pre-built binaries not found for grpc@1.21.1 and node@13.1.0 (node-v79 ABI, glibc) (falling back to source compile with node-gyp)
make: Entering directory '/node_modules/grpc/build'
...
CXX(target) Release/obj.target/gpr/deps/grpc/src/core/lib/profiling/basic_timers.o
CXX(target) Release/obj.target/gpr/deps/grpc/src/core/lib/profiling/stap_timers.o
AR(target) Release/obj.target/libgpr.a
COPY Release/libgpr.a
CXX(target) Release/obj.target/grpc_node/ext/byte_buffer.o
CXX(target) Release/obj.target/grpc_node/ext/call.o
../ext/call.cc: In function 'bool grpc::node::CreateMetadataArray(v8::Local<v8::Object>, grpc_metadata_array*)':
../ext/call.cc:99:58: error: no matching function for call to 'v8::Array::Get(unsigned int&)'
Local<String> current_key(Nan::To<String>(keys->Get(i)).ToLocalChecked());
^
In file included from /root/.cache/node-gyp/13.1.0/include/node/node.h:63:0,
from ../ext/call.cc:23:
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate expects 2 arguments, 1 provided
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate expects 2 arguments, 1 provided
../ext/call.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE grpc::node::Call::StartBatch(Nan::NAN_METHOD_ARGS_TYPE)':
../ext/call.cc:663:35: error: no matching function for call to 'v8::Object::Get(uint32_t&)'
if (!op->ParseOp(obj->Get(type), &ops[i])) {
^
In file included from /root/.cache/node-gyp/13.1.0/include/node/node.h:63:0,
from ../ext/call.cc:23:
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate expects 2 arguments, 1 provided
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate expects 2 arguments, 1 provided
At global scope:
cc1plus: error: unrecognized command line option '-Wno-cast-function-type' [-Werror]
cc1plus: all warnings being treated as errors
make: *** [Release/obj.target/grpc_node/ext/call.o] Error 1
grpc_node.target.mk:186: recipe for target 'Release/obj.target/grpc_node/ext/call.o' failed
make: Leaving directory '/node_modules/grpc/build'
```

> On 25 November 2019 at 16:22 Pat Gavlin <notifications@github.com> wrote:
> 
> 
>     @danhawkes https://github.com/danhawkes can you copy/paste the error you're seeing?
> 
>     —
>     You are receiving this because you authored the thread.
>     Reply to this email directly, view it on GitHub https://github.com/pulumi/pulumi-kubernetes/issues/890?email_source=notifications&email_token=AAZ3LYK5VJVYF4R7YIXBYQDQVP3UBA5CNFSM4JRAOENKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFC6WEQ#issuecomment-558230290 , or unsubscribe https://github.com/notifications/unsubscribe-auth/AAZ3LYOE6M6A7K4YRB2PPOTQVP3UBANCNFSM4JRAOENA .
>

```
FROM node:13
RUN curl -fsSL https://get.pulumi.com | sh
RUN npm i @pulumi/kubernetes
```

```
Step 1/3 : FROM node:13
---> a76ab9b9033f
Step 2/3 : RUN curl -fsSL https://get.pulumi.com | sh
---> Using cache
---> 14d7e19487ed
Step 3/3 : RUN npm i @pulumi/kubernetes
---> Running in 19f50086fc95
> deasync@0.1.16 install /node_modules/deasync
> node ./build.js
`linux-x64-node-13` exists; testing
Binary is fine; exiting
> grpc@1.21.1 install /node_modules/grpc
> node-pre-gyp install --fallback-to-build --library=static_library
node-pre-gyp WARN Using needle for node-pre-gyp https download
node-pre-gyp WARN Tried to download(404): https://node-precompiled-binaries.grpc.io/grpc/v1.21.1/node-v79-linux-x64-glibc.tar.gz
node-pre-gyp WARN Pre-built binaries not found for grpc@1.21.1 and node@13.1.0 (node-v79 ABI, glibc) (falling back to source compile with node-gyp)
make: Entering directory '/node_modules/grpc/build'
...
CXX(target) Release/obj.target/gpr/deps/grpc/src/core/lib/profiling/basic_timers.o
CXX(target) Release/obj.target/gpr/deps/grpc/src/core/lib/profiling/stap_timers.o
AR(target) Release/obj.target/libgpr.a
COPY Release/libgpr.a
CXX(target) Release/obj.target/grpc_node/ext/byte_buffer.o
CXX(target) Release/obj.target/grpc_node/ext/call.o
../ext/call.cc: In function 'bool grpc::node::CreateMetadataArray(v8::Local<v8::Object>, grpc_metadata_array*)':
../ext/call.cc:99:58: error: no matching function for call to 'v8::Array::Get(unsigned int&)'
Local<String> current_key(Nan::To<String>(keys->Get(i)).ToLocalChecked());
^
In file included from /root/.cache/node-gyp/13.1.0/include/node/node.h:63:0,
from ../ext/call.cc:23:
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate expects 2 arguments, 1 provided
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate expects 2 arguments, 1 provided
../ext/call.cc: In static member function 'static Nan::NAN_METHOD_RETURN_TYPE grpc::node::Call::StartBatch(Nan::NAN_METHOD_ARGS_TYPE)':
../ext/call.cc:663:35: error: no matching function for call to 'v8::Object::Get(uint32_t&)'
if (!op->ParseOp(obj->Get(type), &ops[i])) {
^
In file included from /root/.cache/node-gyp/13.1.0/include/node/node.h:63:0,
from ../ext/call.cc:23:
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, v8::Local<v8::Value>)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3471:43: note: candidate expects 2 arguments, 1 provided
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate: v8::MaybeLocal<v8::Value> v8::Object::Get(v8::Local<v8::Context>, uint32_t)
V8_WARN_UNUSED_RESULT MaybeLocal<Value> Get(Local<Context> context,
^~~
/root/.cache/node-gyp/13.1.0/include/node/v8.h:3474:43: note: candidate expects 2 arguments, 1 provided
At global scope:
cc1plus: error: unrecognized command line option '-Wno-cast-function-type' [-Werror]
cc1plus: all warnings being treated as errors
make: *** [Release/obj.target/grpc_node/ext/call.o] Error 1
grpc_node.target.mk:186: recipe for target 'Release/obj.target/grpc_node/ext/call.o' failed
make: Leaving directory '/node_modules/grpc/build'
```

--- a/pkg/gen/nodejs-templates/package.json.mustache
+++ b/pkg/gen/nodejs-templates/package.json.mustache
@@ -13,7 +13,7 @@
         "build": "tsc"
     },
     "dependencies": {
-        "@pulumi/pulumi": "1.5.0-alpha.1572979668",
+        "@pulumi/pulumi": "^1.6.0",
         "@types/js-yaml": "^3.11.2",
         "js-yaml": "^3.12.0",
         "shell-quote": "^1.6.1",

--- a/sdk/nodejs/package.json
+++ b/sdk/nodejs/package.json
@@ -13,7 +13,7 @@
         "build": "tsc"
     },
     "dependencies": {
-        "@pulumi/pulumi": "1.5.0-alpha.1572979668",
+        "@pulumi/pulumi": "^1.6.0",
         "@types/js-yaml": "^3.11.2",
         "js-yaml": "^3.12.0",
         "shell-quote": "^1.6.1",

